presto-parser/src/main/java/com/facebook/presto/sql/tree/QualifiedName.java

```diff
@@ -32,6 +32,9 @@
     private final List<String> parts;
     private final List<String> originalParts;
 
+    private Optional<QualifiedName> cachedPrefix;
+    private String cachedSuffix;
```

Comment:
This class is meant to be immutable and thread-safe. This change breaks that assumption.